main.go

```diff
@@ -48,6 +50,8 @@ func main() {
 	flagset := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
 	flagset.StringVar(&insecureListenAddress, "insecure-listen-address", "", "The address the prom-label-proxy HTTP server should listen on.")
 	flagset.StringVar(&internalListenAddress, "internal-listen-address", "", "The address the internal prom-label-proxy HTTP server should listen on to expose metrics about itself.")
+	flagset.StringVar(&queryParam, "query-param", "", "Name of the GET query param to define the tenant. If empty and header-name is empty, the value of label will be used.")
```

Comment:
The `-label-value` description would need an update to say "the label value will be taken from the URL parameter ... or from the HTTP header ...".
Also it would be good to mark `--label`  as deprecated.

Reply:
> Also it would be good to mark --label as deprecated.

Why? Its still used for configure the restriction label on PromQL side.

Reply:
right my bad!